src/libcommon/log/sentry/handler.cpp

```diff
@@ -363,54 +371,58 @@ void SentryHandler::handleEventsRateLimit(SentryEvent &event, bool &toUpload) {
     escalateSentryEvent(event);
 }
 
-bool SentryHandler::lastEventCaptureIsOutdated(const SentryEvent &event) const {
+bool Handler::lastEventCaptureIsOutdated(const SentryEvent &event) const {
     using namespace std::chrono;
     return (event.lastCapture + seconds(_sentryMinUploadIntervaOnRateLimit)) <= system_clock::now();
 }
 
-bool SentryHandler::lastEventUploadIsOutdated(const SentryEvent &event) const {
+bool Handler::lastEventUploadIsOutdated(const SentryEvent &event) const {
     using namespace std::chrono;
     return (event.lastUpload + seconds(_sentryMinUploadIntervaOnRateLimit)) <= system_clock::now();
 }
 
-void SentryHandler::escalateSentryEvent(SentryEvent &event) {
+void Handler::escalateSentryEvent(SentryEvent &event) const {
     event.message += " (Rate limit reached: " + std::to_string(event.captureCount) + " captures.";
-    if (event.level == SentryLevel::Fatal || event.level == SentryLevel::Error) {
+    if (event.level == Level::Fatal || event.level == Level::Error) {
         event.message += ")";
         return;
     }
 
     event.message += " Level escalated from " + toString(event.level) + " to Error)";
-    event.level = SentryLevel::Error;
+    event.level = Level::Error;
 }
 
-void SentryHandler::updateEffectiveSentryUser(const SentryUser &user) {
+void Handler::updateEffectiveSentryUser(const SentryUser &user) {
     if (_globalConfidentialityLevel == _lastConfidentialityLevel && user.isDefault()) return;
     _lastConfidentialityLevel = user.isDefault() ? _globalConfidentialityLevel : SentryConfidentialityLevel::Specific;
 
     sentry_value_t userValue;
-    if (!user.isDefault()) {
+    if (_globalConfidentialityLevel == SentryConfidentialityLevel::Anonymous) {
```

Comment:
When does it apply?

Reply:
For the moment, it is never used, but it is ready to be used when we set up a modal to allow the user to choose which data they accept to send.